### DISCUSSION TITLE
How to set the classic Macintosh encoding of the font?

### DISCUSSION THREAD

**Discussion by xuhao148** [fontforge/fontforge#5466] [Q&A] [answered]
I want to use FontForge to convert a Japanese OpenType font to .suit format so I can use it on my old Mac. I imported the font, CID-flattened it, reencoded it to SJIS and clicked Generate Font, however when I opened it on Mac OS 9.2 the sample text was in English (_cozy lummox_ things), and I believe Mac OS was treating this font as a Mac Roman encoding font as Japanese text showed up as accented Latin characters. I looked through the Font Info dialog but did not find anything special; the Mac tab only allows me to change the FOND resource name and font variant.
Where can I set the classic Macintosh encoding of the font? Thanks.

**Comment by skef** [comment]
I did a little research on this. System 9 had some support for Unicode but most applications couldn't make use of it. The support for Japanese and a number of other languages appears to have been added in System 7.1 via a system called "WorldScript", with specific languages added by installing "Language Kits". 

There is not a lot of information about these systems that I could find. The most useful resource may be this on on chinese: https://chinesemac.org/pages/old/os8.html . It points to https://web.archive.org/web/20050831162921/http://ftp.nsysu.edu.tw/Mac/Chinese/classic/font/ , which has a list of "NiceTaipei" font files, those being better (at the time) replacements for the Taipei system font that came with the language kit. The .sit.hqx file is not there, but the .sit.bin at the bottom is. If you're running classic macos you presumably have some version of StuffIt. Or if not "unar" on Linux can decompress these files. Looking at the file may, or may not, give you some idea of what you actually have to build. WorldScript seems to predate Unicode, so there's no strong evidence that you can just build a Unicode font. And I see no evidence that FontForge will build a custom "two byte" Macintosh font from this era.

**Comment by skef** [comment]
Also found [aparkin,+23428-64426-1-PB.pdf](https://github.com/user-attachments/files/16921611/aparkin.%2B23428-64426-1-PB.pdf). Looks like the encoding you're probably looking for is indicated by kMacJapaneseStandardVariant or kMacJapaneseVertAtKuPlusTenVariant (https://developer.apple.com/documentation/coreservices/1399915-encoding_variants_for_macjapanes?changes=l_9 ). The former is described as "Shift-JIS with JIS Roman modifications, extra 1-byte characters, 2-byte Apple extensions, and some vertical presentation forms in the range 0xEB40-0xEDFE ("ku plus 84")"

**Comment by skef** [comment]
This might be the encoding mapping to unicode: https://unicode.org/Public/MAPPINGS/VENDORS/APPLE/JAPANESE.TXT .

**Comment by skef** [comment]
OK, *maybe* what you're looking to do is produce a TrueType font with a cmap that uses 1 (Macintosh) as the platformID and 1 (Japanese) as the platformSpecificID. ( https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6name.html ) and with the glyphs encoded like in that description from the last message. Then you would translate that font to resource fork format, probably following some of the advice https://68kmla.org/bb/index.php?threads/truetype-fonts-under-system-7.40887/ rather than trying to use fontforge for it.

**Comment by skef** [comment, marked as answer]
OK, now that I think about it, those are close enough to Shift JIS that we get back to your question. 

If you're working with a TrueType data file, the thing to do is to dump it with ttx (in the fonttools python package) and look at the cmap, (the formats, the platformid and platformspecificid, and the actual encoding). If anything is off you can try editing the xml and running ttx to get back a binary file.

**Reply by xuhao148** [reply]
So I guess I need to write a script to convert the original cmap format 4 mapping Unicode to a cmap format 2 mapping Mac Japanese, so that the old Mac OS can recognize it. Thank you!
